### ISSUE TITLE
ncurses recursive symlink

### ISSUE THREAD

**Issue by mboisson** [NixOS/nixpkgs#34472]
## Issue description

The ncurses recipe creates symlinks to the parent directory for "include/ncurses" and "include/ncursesw". This is to ensure that one can include header files using `ncurses/<header>`, `ncursesw/<header>` or just `<header>`. The problem is that when you try to install both ncurses5 and ncurses, using nixos 16.09, these symlinks send `nix-env --preserve-installed -i` into an infinite loop. This seems to be fixed in 17.09, but we run 16.09. 

In our fork of nixpkgs, I have this commit: 
https://github.com/ComputeCanada/nixpkgs/commit/040fa5b27c2f1b0222693ae36b92e87efa377063

This replaces the symlink of the directory by symlinks of each header file. This is the strategy used on CentOS and Fedora, and it prevents the problem I ran into.

My question: Do people care and should I open a PR upstream for this commit (and if so against which branch, 16.09 ? 17.09 ? other ?)

**Comment by pbogdan** [comment]
With the disclaimer that I don't know anything about your particular issue - I believe 16.09 is not supported any more and is not receiving any updates. At the moment the only supported stable release is 17.09.

**Comment by JohnAZoidberg** [comment]
It's 2019 now.

**Comment by twhitehead** [comment]
If I understand the origin issue was that nix-env wasn't able to properly combine the include symlinks for multiple ncurses

```
$ ls -l /nix/store/*ncurses*-dev/include 
/nix/store/wa4p4xm8b6m3hdypp7z2mwwcqk6a2f58-ncurses-6.1-20190112-dev/include:
...
lrwxrwxrwx 1 root root     1 Dec 31  1969 ncurses -> .
lrwxrwxrwx 1 root root     8 Dec 31  1969 ncurses.h -> curses.h
lrwxrwxrwx 1 root root     1 Dec 31  1969 ncursesw -> .
```

and a solution was to instead (of symlinking _ncurses_ and _ncursesw_ to _._) make _ncurses_ and _ncursesw_ directories that contained symlinks back to all the individual header files (as is done in fedora and such).

As per mboisson's comment, it seems this was fixed in 17.09 though, so I think we might as well close it as 16.09 is truly old now.
